### PR TITLE
Rvizdev

### PR DESCRIPTION
**Pull request by Riteshlogin**
Not much to say, let me know if there is anything I should change or do.

### CODE REVIEW

File: src/basic_shapes.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/*
```
Comment (aftersomemath) [inline src/basic_shapes.cpp]: Do we need to include this file?

File: src/rviz_publisher.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    uint32_t shape = visualization_msgs::Marker::CUBE;
+    marker.header.frame_id = "/my_frame";
+    marker.header.stamp = ros::Time::now();
+    marker.ns = "basic_shapes";
+    marker.id = 0;
+    marker.type = shape;
+    marker.pose.position.x = 0;
+    marker.pose.position.y = 0;
+    marker.pose.position.z = 0;
+    double t0 = std::cos(msg->data.yaw * 0.5f);
```
Comment (aftersomemath) [inline src/rviz_publisher.cpp]: Has this conversion been tested? I'd rather use the built in tf2 conversion capabilities in the future but if this works already it's fine.

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  <!-- Use run_depend for packages you need at runtime: -->
+  <!--   <run_depend>message_runtime</run_depend> -->
+  <!-- Use test_depend for packages you need only for testing: -->
+  <!--   <test_depend>gtest</test_depend> -->
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>iarc7_msgs</build_depend>
+  <build_depend>roscpp</build_depend>
+  <build_depend>rosconsole</build_depend>
+  <run_depend>rviz</run_depend>
+  <run_depend>tf2</run_depend>
```
Comment (aftersomemath) [inline package.xml:49]: Do we need these run dependencies? I see that the build dependencies were removed.

File: src/rviz_publisher.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    marker.lifetime = ros::Duration();
+
+    marker_pub.publish(marker);
+
+
+  }
+
+private:
+  ros::NodeHandle n; 
+  ros::Publisher marker_pub;
```
Comment (aftersomemath) [inline src/rviz_publisher.cpp:89]: Can we give these a name specific to the type of thing they are subscribing or publishing too? That will help when we start to add more publishers and subscribers.

File: package.xml (hunk 3 above)
Comment (Riteshlogin) [reply package.xml:49]: We need rviz and rosconsole for sure, did you mean tf2/tf2_ros? (I also saw what you said about using the tf2 conversion and I think I will try to utilize that)

Comment (Riteshlogin) [inline package.xml]: http://docs.ros.org/jade/api/tf2/html/classtf2_1_1Quaternion.html#a4c9249826320a3aa30e88a11fc75e577

You're right, using tf2 seems a lot easier

Comment (aftersomemath) [reply package.xml:49]: I merely meant rviz and tf2. If you use tf2 then I think you should have the run dependency and you definitely need a build dependency for it too.

**Review by amiller27 — APPROVED** [review]

**Pull request merged**